lib/rubygems/specification.rb

```diff
@@ -350,7 +350,8 @@ def authors=(value)
   # You should specify a license for your gem so that people know how they are
   # permitted to use it and any restrictions you're placing on it.  Not
   # specifying a license means all rights are reserved; others have no right
-  # to use the code for any purpose.
+  # to use the code for any purpose. You can also specifically set the license
+  # to nil to specify all rights are reserved.
```

Comment:
I missed this discussion so forgive me if this was already discussed.

I don't think we should give a meaning to nil. Nil is nil, no license. The meaning of that is unknown without more information. I would therefore remove this comment and any other text that might imply some (legal-like) meaning from `nil`.

If you want an all right reserved license, would it be reasonable to instead add that specifically?

Reply:
Discussion is at https://github.com/rubygems/rubygems/issues/6724. My understanding from the error message is that license is supposed to be filled with an opensource license. There for absence of a license means the gem could be potentially private. The whole point is to allow the warning about missing license to be avoided in this case.

Thinking about it again, you're right that adding a special value for this case may be better. Maybe the whole problem is that the current "Nonstandard" special value is not good enough for private gem, since "Nonstandard" was probably meant for Open Source licenses that have not yet been standardized. So we could add a different special value for private gems.

Reply:
Thanks for the review folks! 

To play devil's advocate, since the convention has been for no license representing all rights reserved, wouldn't `nil` keep with that convention compared to a new specific value for private gems? As in the `nil` case, the devs still haven't specified a license.

Reply:
[The first google result](https://choosealicense.com/no-permission/#:~:text=When%20you%20make%20a%20creative,shake%2Ddowns%2C%20or%20litigation.) says similar to what we've arrived at here. No license just means you get whatever happens when you don't specify a license.

The simplest solution here is "nil removes the error and makes no claim about any license, non license, or otherwise." I suggest simply removing the extra checking and allowing nil. If you specify nil and MIT, then you get MIT because nil makes no claim about licensing.